ccore/nebula/client_meta.go

```
@@ -31,48 +34,151 @@ func NewMetaClient(endpoints []string, opts ...Option) (MetaClient, error) {
+		resp, err = c.meta.Balance(types.BalanceReq{
+			Cmd:   types.BalanceData,
+			Space: space,
+		})
+		return resp, err
 	})
+
+	if retryErr != nil {
+		return nil, retryErr
+	}
+
+	return
 }
 
-func (c *defaultMetaClient) BalanceLeader(space string) (types.Balancer, error) {
-	return c.meta.Balance(types.BalanceReq{
-		Cmd:   types.BalanceLeader,
-		Space: space,
+func (c *defaultMetaClient) BalanceLeader(space string) (resp types.Balancer, err error) {
+	retryErr := c.retryDo(func() (types.MetaBaser, error) {
+		resp, err = c.meta.Balance(types.BalanceReq{
+			Cmd:   types.BalanceLeader,
+			Space: space,
+		})
+		return resp, err
 	})
+
+	if retryErr != nil {
+		return nil, retryErr
+	}
+
+	return
 }
 
-func (c *defaultMetaClient) BalanceDataRemove(space string, endpoints []string) (types.Balancer, error) {
-	return c.meta.Balance(types.BalanceReq{
-		Cmd:           types.BalanceDataRemove,
-		Space:         space,
-		HostsToRemove: endpoints,
+func (c *defaultMetaClient) BalanceDataRemove(space string, endpoints []string) (resp types.Balancer, err error) {
+	retryErr := c.retryDo(func() (types.MetaBaser, error) {
+		resp, err = c.meta.Balance(types.BalanceReq{
+			Cmd:           types.BalanceDataRemove,
+			Space:         space,
+			HostsToRemove: endpoints,
+		})
+		return resp, err
 	})
+
+	if retryErr != nil {
+		return nil, retryErr
+	}
+
+	return
 }
 
 func (c *defaultMetaClient) defaultClient() *defaultClient {
 	return (*defaultClient)(c)
 }
+
+func (c *defaultMetaClient) retryDo(fn func() (types.MetaBaser, error)) error {
+	resp, err := fn()
+	if err != nil {
+		// check if transport exception
+		if err = c.reconnect(err); err != nil {
+			return err
+		}
+		resp, err = fn()
+		if err != nil {
+			return err
+		}
+	}
+	// check if leader change
+	if resp.GetCode() == nerrors.ErrorCode_E_LEADER_CHANGED {
+
+		if err = c.updateLeader(resp.GetLeader()); err != nil {
+			return err
+		}
+		_, _ = fn()
```

Why ignore this error, but do not ignore it in line 144?